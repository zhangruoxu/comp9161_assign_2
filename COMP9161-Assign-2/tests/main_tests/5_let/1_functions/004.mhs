main
     =
    let f =
           letfun f x = -1;
    in let f  =
           letfun f x = -2;
    in let f  =
           letfun f  x = -3;
    in let f =
           letfun f x = -4;
    in let f =
           letfun f x = -5;
    in let f =
           letfun f x = -6;
    in let f =
           letfun f x = -7;
    in let f =
           letfun f x = -8;
    in let f =
           letfun f x = -9;
    in let f =
           letfun f x = True;
    in f 7;
