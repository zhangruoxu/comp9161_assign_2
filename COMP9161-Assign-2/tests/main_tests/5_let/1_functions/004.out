main :: Bool = let f :: forall a. (a -> Int) = (letfun f :: (a -> Int) x = (- 1));
               in let f :: forall d. (d -> Int) = (letfun f :: (d -> Int) x = (- 2));
                  in let f :: forall g. (g -> Int) = (letfun f :: (g -> Int) x = (- 3));
                     in let f :: forall j. (j -> Int) = (letfun f :: (j -> Int) x = (- 4));
                        in let f :: forall m. (m -> Int) = (letfun f :: (m -> Int) x = (- 5));
                           in let f :: forall p. (p -> Int) = (letfun f :: (p -> Int) x = (- 6));
                              in let f :: forall s. (s -> Int) = (letfun f :: (s -> Int) x = (- 7));
                                 in let f :: forall v. (v -> Int) = (letfun f :: (v -> Int) x = (- 8));
                                    in let f :: forall y. (y -> Int) = (letfun f :: (y -> Int) x = (- 9));
                                       in let f :: forall a2. (a2 -> Bool) = (letfun f :: (a2 -> Bool) x = True);
                                          in f 7;
